### DISCUSSION TITLE
Failed install at step 6 "lib mamba Could not solve for environment specs"

### DISCUSSION THREAD

**Discussion by pickleton89** [biostars/biostar-handbook-issues#300] [General]
I keep running into this error when running the:
`curl http://data.biostarhandbook.com/install.sh | bash`

# 6 Installing tools into bioinfo
#
warning  libmamba Problem type not implemented SOLVER_RULE_STRICT_REPO_PRIORITY
warning  libmamba Problem type not implemented SOLVER_RULE_STRICT_REPO_PRIORITY
warning  libmamba Problem type not implemented SOLVER_RULE_STRICT_REPO_PRIORITY
warning  libmamba Problem type not implemented SOLVER_RULE_STRICT_REPO_PRIORITY
warning  libmamba Problem type not implemented SOLVER_RULE_STRICT_REPO_PRIORITY
warning  libmamba Problem type not implemented SOLVER_RULE_STRICT_REPO_PRIORITY
warning  libmamba Problem type not implemented SOLVER_RULE_STRICT_REPO_PRIORITY
warning  libmamba Problem type not implemented SOLVER_RULE_STRICT_REPO_PRIORITY
warning  libmamba Problem type not implemented SOLVER_RULE_STRICT_REPO_PRIORITY
warning  libmamba Problem type not implemented SOLVER_RULE_STRICT_REPO_PRIORITY
error    libmamba Could not solve for environment specs
    The following packages are incompatible
    └─ samtools >=1.14  is not installable because there are no viable options
       ├─ samtools [1.14|1.15] would require
       │  └─ htslib >=1.14,<1.19.0a0 , which conflicts with any installable versions previously reported;
       ├─ samtools [1.15|1.15.1] would require
       │  └─ htslib >=1.15,<1.19.0a0 , which conflicts with any installable versions previously reported;
       ├─ samtools [1.15.1|1.16.1] would require
       │  └─ htslib >=1.16,<1.19.0a0 , which conflicts with any installable versions previously reported;
       ├─ samtools [1.16.1|1.17] would require
       │  └─ htslib >=1.17,<1.19.0a0 , which conflicts with any installable versions previously reported;
       ├─ samtools [1.17|1.18] would require
       │  └─ htslib >=1.17,<1.18.0a0 , which conflicts with any installable versions previously reported;
       └─ samtools 1.18 would require
          └─ htslib >=1.18,<1.19.0a0 , which conflicts with any installable versions previously reported.
critical libmamba Could not solve for environment specs

I installed samtools and htslib with the following command:
`micromamba install samtools`
`micromamba install htslib`

The errors still exists.

**Comment by ialbert** [comment]
what operating system is this?

did you have conda already installed? 

Sometimes if you did already have `conda` there the previous `.condarc` could be incorrect

the `~/.condarc` file should only contain:

```
channels:
  - conda-forge
  - bioconda
channel_priority: strict
```

**Reply by pickleton89** [reply]
Sorry, I forgot to list that M1 MAC Sonoma 14.1. This is what was in my `~/.condarc` file.
---
channels:
  - conda-forge
  - defaults
  - bioconda
offline: false
channel_priority: strict
---

I did have a previous install of biostar materials, but wanted a reboot as my organization had got away from me.

**Reply by ialbert** [reply]
replace your `.condarc` with the one I showed

**Reply by ialbert** [reply]
or delete your `.condarc` and the installer will recreate it with the correct layout.

**Comment by ialbert** [comment]
Turns out I could also force the correct order in the `install.sh` so I have made the change that next time will correct the order of channels even if these previously existed in the wrong order. 

You can also do it manually with:

```
micromamba config prepend channels bioconda
micromamba config prepend channels conda-forge
```

**Reply by pickleton89** [reply]
Thanks. I deleted the `.condarc` file as you suggested. I am currently installing, I am at step 6, and appears to be working as no error at this time.

**Reply by pickleton89** [reply]
doctor.py said that I am doing well :)

Thank you for your assistance.

**Reply by ialbert** [reply]
Thanks for the report, I fixed the `install.sh` so it would correct this inconsistency in the future.

**Comment by xiangyuanyu** [comment]
error    libmamba Could not solve for environment specs
    The following packages are incompatible
    └─ samtools >=1.14 * is not installable because there are no viable options
       ├─ samtools [1.14|1.15] would require
       │  └─ htslib >=1.14,<1.23.0a0 *, which conflicts with any installable versions previously reported;
       ├─ samtools [1.15|1.15.1] would require
       │  └─ htslib >=1.15,<1.23.0a0 *, which conflicts with any installable versions previously reported;
       ├─ samtools [1.15.1|1.16.1] would require
       │  └─ htslib >=1.16,<1.23.0a0 *, which conflicts with any installable versions previously reported;
       ├─ samtools [1.16.1|1.17|1.18] would require
       │  └─ htslib >=1.17,<1.23.0a0 *, which conflicts with any installable versions previously reported;
       ├─ samtools 1.18 would require
       │  └─ htslib >=1.18,<1.23.0a0 *, which conflicts with any installable versions previously reported;
       ├─ samtools 1.19 would require
       │  └─ htslib >=1.19,<1.23.0a0 *, which conflicts with any installable versions previously reported;
       ├─ samtools [1.19.1|1.19.2] would require
       │  └─ htslib >=1.19.1,<1.23.0a0 *, which conflicts with any installable versions previously reported;
       ├─ samtools 1.20 would require
       │  └─ htslib >=1.20,<1.23.0a0 *, which conflicts with any installable versions previously reported;
       ├─ samtools 1.21 would require
       │  └─ htslib >=1.21,<1.23.0a0 *, which conflicts with any installable versions previously reported;
       ├─ samtools 1.22 would require
       │  └─ htslib >=1.22,<1.23.0a0 *, which conflicts with any installable versions previously reported;
       └─ samtools 1.22.1 would require
          └─ htslib >=1.22.1,<1.23.0a0 *, which conflicts with any installable versions previously reported.
critical libmamba Could not solve for environment specs

yumnera@Mac ~
$ cat ~/.condarc
channels:
  - conda-forge
  - bioconda
channel_priority: strict
I met with similar issue, but after I check the condarc, and find out the correct order. I am using Mac os 15.0, M2 chip. I wonder if you could help me solve that question, thank you!

**Comment by ialbert** [comment]
This sounds like you have an M processor-based micromamba rather than the Intel-based micromamba.

Even on M type processor you have to run the Intel based  micromamba.

By default the install script will install the correct version but if you already had micromamba or conda installed issues might arise.

**Reply by xiangyuanyu** [reply]
So how should I uninstall them and reset the new envir?

**Reply by ialbert** [reply]
yes, the enviroment needs to be Intel based, please follow the instructions in the book, it describes all steps in detail
